ansible/playbooks/roles/postgresql/tasks/helpers/configure-ssh-access.yml

```diff
@@ -0,0 +1,73 @@
+---
+- name: Ensure openssh packages are installed
+  package:
+    name: "{{ _packages[ansible_os_family] }}"
+    state: present
+  vars:
+    _packages:
+      Debian:
+        - openssh-client
+        - openssh-server
+      RedHat:
+        - openssh-clients
+        - openssh-server
+
+- name: Ensure sshd is started and enabled
+  systemd:
+    name: sshd
+    enabled: true
+    state: started
+
+- name: Set getent_passwd fact
+  getent:
+    database: passwd
+    key: "{{ username }}"
+    split: ":"
+
+- become_user: "{{ username }}"
+  vars:
+    home_dir: "{{ ansible_facts.getent_passwd[username][4] }}"
+  block:
+    - name: Ensure {{ home_dir }}/.ssh directory exists
+      file:
+        path: "{{ home_dir }}/.ssh/"
+        state: directory
+        mode: u=rwx,go=
+
+    - name: Generate openssh keypair
+      openssh_keypair:
+        path: "{{ home_dir }}/.ssh/id_ed25519"
+        type: ed25519
+        regenerate: full_idempotence
```

Comment:
Did you consider to create backup of existing key?

"If set to ```full_idempotence```, the key will be regenerated if it does not conform to the module's options. This is also the case if the key cannot be read (broken file), the key is protected by an unknown passphrase, or when they key is not protected by a passphrase, but a passphrase is specified. Make sure you have a ```backup``` when using this option!" [DOC](https://docs.ansible.com/ansible/latest/collections/community/crypto/openssh_keypair_module.html#parameter-regenerate)